src/Components/Blazor/Templates/src/content/BlazorWasm-CSharp/Client/App.razor

```diff
@@ -1,5 +1,10 @@
-﻿<Router AppAssembly="typeof(Program).Assembly">
+﻿<Router AppAssembly="@typeof(Program).Assembly">
+    <Found Context="routeData">
+        <RouteView RouteData="@routeData" DefaultLayout="@typeof(MainLayout)" />
+    </Found>
     <NotFound>
-        <p>Sorry, there's nothing at this address.</p>
+        <LayoutView Layout="@typeof(MainLayout)">
+            <p>Sorry, there's nothing at this address.</p>
+        </LayoutView>
```

Comment:
The no-auth case for this seems really simple and cool. 

Something we could do (that would trade one kind of complexity for another) would be to add a `<DefaultLayout>` as a cascading value provider component. 

I know you tend to prefer using `Context="..."` but I tend to not. Ultimately we could end up with something like:
```razor
<DefaultLayout Layout="typeof(MainLayout)">
<Router AppAssembly="@typeof(Program).Assembly">
    <Found>
        <RouteView RouteData="@context">
    </Found>
    <NotFound>
        <LayoutView>
            <p>Sorry, there's nothing at this address.</p>
        </LayoutView>
    </NotFound>
</Router>
</DefaultLayout>
```

In these case there's very little duplication and noise. These are nitpicks though, I'm happy with the result. 

I'm not sure it's the *right thing* to use a cascasing value just to DRY-up the code, but it can make it less verbose.

Reply:
Sure, I was definitely aware of this option but sided against it. I'm concerned with trying to make `<LayoutView>` generally useful in more scenarios than only the top-level `App.razor` file. It would be totally reasonable to use `<LayoutView>` at multiple levels, especially if we have nested routers in the future.

In cases where `<LayoutView>` is used outside `App.razor`, you really want to make it use the right layout for its own level, and not randomly inherit a choice from further up the tree (from some other file you can't necessarily find because there's no F12 way of getting to it). It seems greatly preferable for it to be a required explicit parameter.

Of course if someone wants, they can make their own `<CascadedLayoutView>` that consumed a cascading parameter and outputs a `<LayoutView>`. The feature is achievable in user code, but I don't fancy it as the built-in default, even if we need slightly more explicitness in `App.razor`.